File: Java/diccionariosycolecciones/Ejercicio2.java
package diccionariosycolecciones;

import java.util.ArrayList;

/**
 * Ejercicio 2:
 * 
 * Realiza un programa que introduzca valores aleatorios (entre 0 y 100) en un
 * ArrayList y que luego calcule la suma, la media, el máximo y el mínimo de esos
 * números. El tamaño de la lista también será aleatorio y podrá oscilar entre 10
 * y 20 elementos ambos inclusive.

 * @author Francisco Javier González Sabariego
 *
 */
public class Ejercicio2 {
  public static void main(String[] args) {
    //Variables:
    
    int tamanoAleatorio = 0;  //El tamaño de la lista (entre 10 y 20)    
    int numAleatorio = 0;     //Número aleatorio (entre 0 y 100) que se almacena en el ArrayList
    int suma = 0;             //Variable que almacena la suma de todos los valores del ArrayList
    int maximo = 0;           //Variable que almacena el valor máximo del Arraylist
    int minimo = 100;         //Variable que almacena el valor mínimo del Arraylist
    
    ArrayList<Integer> alea = new ArrayList<Integer>();
    
    
    //Definimos el tamaño del array List:
    tamanoAleatorio = (int)(Math.random()*10)+11;
    
    //Insertamos valores en el arrayList:
    for (int i=0; i<tamanoAleatorio; i++) {
      numAleatorio = (int)(Math.random()*101);
      alea.add(numAleatorio);
    }
    
    System.out.print("\n\nEste es el ArrayList: \n");
    
    //Mostramos el arrayList:
    for (int i=0; i<tamanoAleatorio; i++) {
      System.out.print(alea.get(i)+"\n");
      
      suma += alea.get(i);            //Sumamos
      
      if (minimo > alea.get(i)) {     //Calculamos mínimo
        minimo = alea.get(i);
      }
      
      if (maximo < alea.get(i)) {     //Calculamos máximo
        maximo = alea.get(i);
      }
    }
    
    System.out.print("\n\nLa suma es: " + suma);
    System.out.print("\nLa media es : " + (double)suma/alea.size());
    System.out.print("\nEl valor máximo es: " + maximo);
    System.out.print("\nEl valor mínimo es: " + minimo);
    
  }
}

File: Python/diccionariosycolecciones/Ejercicio2.py
'''
Created on 23 mar. 2019

 * Ejercicio 2:
 * 
 * Realiza un programa que introduzca valores aleatorios (entre 0 y 100) en un
 * ArrayList y que luego calcule la suma, la media, el máximo y el mínimo de esos
 * números. El tamaño de la lista también será aleatorio y podrá oscilar entre 10
 * y 20 elementos ambos inclusive.

@author: Francisco Javier González Sabariego
'''

import random

lista = list()
aleatorio = int((random.random()*10)+11)
suma = 0
minimo=1000
maximo=0

for i in range(aleatorio+1):
    a=int(random.random()*101)
    lista.append(a)

lista.sort(key=None, reverse=False)

for i in range (aleatorio+1):
    print(lista[i])
    suma += lista[i]
    if minimo>lista[i]:
        minimo=lista[i]
    if maximo<lista[i]:
        maximo=lista[i]

print("\nLa suma es: ", end="")
print(suma)
print("La media es: ", end="")
print(suma/len(lista))
print("El mínimo es: ", end="")
print(minimo)
print("El máximo es: ", end="")
print(maximo)

